packages/exoflex/src/components/RichRadio/RichRadioGroup.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode } from 'react';
+import {
+  StyleSheet,
+  StyleProp,
+  ViewStyle,
+  ScrollView,
+  View,
+} from 'react-native';
+
+import useTheme from '../../helpers/useTheme';
+
+export type RichRadioGroupProps<T> = {
+  data: Array<T>;
+  separatorWidth?: number;
+  keyExtractor: (item: T, index: number) => string;
+  renderItem: ({ item, index }: { item: T; index: number }) => ReactNode;
+  style?: StyleProp<ViewStyle>;
+  contentContainerStyle?: StyleProp<ViewStyle>;
+  testID?: string;
+};
+
+export default function RichRadioGroup<T>(props: RichRadioGroupProps<T>) {
+  let {
+    data,
+    separatorWidth,
+    keyExtractor,
+    renderItem,
+    style,
+    contentContainerStyle,
+    testID,
+  } = props;
+  let { style: themeStyle } = useTheme();
+
+  let rootStyle = [themeStyle?.richRadioGroup?.style, style] as ViewStyle;
+  let containerStyle = [
+    styles.container,
+    themeStyle?.richRadioGroup?.contentContainerStyle,
+    contentContainerStyle,
+  ] as ViewStyle;
+
+  return (
+    <ScrollView
+      horizontal
+      showsHorizontalScrollIndicator={false}
+      style={rootStyle}
+      contentContainerStyle={containerStyle}
```

Comment:
Hmm, what happens if the user wants a vertical list of a radio button?

Probably it won't be as simple as `flex-direction: column`.

Reply:
Also, I'm not 100% sure about using scroll view here.
Depending on the design, the radio item would wrap on a new line or column depending on the direction instead of scroll-able.

Or for that case, we left the user to build their own wrapper and only provide scroll-able group from exoflex?

Reply:
I've discussed about that with Vincent before. And for this component, he said there is a limitation that it could only horizontally scrollable.

So this component won't have vertical list ability as that is what our normal radio button does. That is one part that makes this component different from our normal radio button.